File: backend/src/main/java/io/github/amandajuchem/projetoapi/entities/Atendimento.java
package io.github.amandajuchem.projetoapi.entities;

import io.github.amandajuchem.projetoapi.enums.Motivo;
import jakarta.persistence.*;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "tb_atendimentos")
public class Atendimento extends AbstractEntity {

    @NotNull
    @Column(name = "data_hora")
    private LocalDateTime dataHora;

    @NotNull
    @Column(name = "motivo", length = 25)
    @Enumerated(EnumType.STRING)
    private Motivo motivo;

    @Column(name = "diagnostico", columnDefinition = "VARCHAR")
    private String diagnostico;

    @Column(name = "posologia", columnDefinition = "VARCHAR")
    private String posologia;

    @Valid
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    private List<ExameRealizado> examesRealizados;

    @OneToOne
    private Animal animal;

    @OneToOne
    private Usuario veterinario;
}

File: frontend/src/app/entities/atendimento.ts
import { AbstractEntity } from './abstract-entity';
import { Animal } from './animal';
import { ExameRealizado } from './exame-realizado';
import { Usuario } from './usuario';

export interface Atendimento extends AbstractEntity {
    dataHora: Date;
    dataHoraRetorno: Date;
    animal: Animal;
    veterinario: Usuario;
    motivo: string;
    diagnostico: string;
    procedimentos: string;
    posologia: string;
    examesRealizados: Array<ExameRealizado>;
}